libraries/chain/eosio_contract.cpp

```diff
@@ -167,19 +167,15 @@ void apply_eosio_setcode(apply_context& context) {
       }
    }
 
-
-   code_object::id_type code_id; // default is 0 which indicates no code is present
    if( code_size > 0 ) {
       const code_object* new_code_entry = db.find<code_object, by_code_hash>(
                                              boost::make_tuple(code_hash, act.vmtype, act.vmversion) );
       if( new_code_entry ) {
          db.modify(*new_code_entry, [&](code_object& o) {
-            code_id = o.id;
             ++o.code_ref_count;
          });
       } else {
          db.create<code_object>([&](code_object& o) {
-            code_id = o.id;
             o.code_hash = code_hash;
             o.code.assign(act.code.data(), code_size);
             o.code_ref_count = 1;
```

Comment:
`o.first_block_used` in line below should be set to `context.control.head_block_num() + 1`.